### DISCUSSION TITLE
Ideas / roadmap for tool deprecations

### DISCUSSION THREAD

**Discussion by bernt-matthias** [galaxyproject/galaxy#19053] [General Discussion]
Current state

- Currently tool repos mark tools as deprecated by moving them to a `deprecated` folder.
- Sometimes the tool is manually marked as deprecated in the TS.
- Sometimes something is done on one or the other Galaxy instance (hiding the tool / adding a tag)

The tool repos should  automatically mark the tools in `deprecated/` as `deprecated` in the TS. 

- [ ] implement in bioblend
- [ ] implement in planemo (`shed_deprecate`?)
- [ ] implement in IUC

Then the Galaxy admins might be able to automatically respond to tool deprecations in an automated manner.

- How could this be done? Modifying the shed_tools config file? API? 

xref https://github.com/galaxyproject/galaxy/discussions/17951

**Comment by martenson** [comment]
quicknote: We need to be mindful here about what `deprecated` in TS means. I am not sure but in addition to non-searchable (which is already a problem) it might mean non-installable.

**Reply by bernt-matthias** [reply]
>  it might mean non-installable.

Maybe. At least they are [non-uninstallable](https://github.com/galaxyproject/galaxy/issues/16258) .. LOL
